unstructured_ingest/v2/processes/connectors/gitlab.py

```
@@ -0,0 +1,249 @@
+                and (
+                    not self.connection_config.git_file_glob
+                    or self.does_path_match_glob(element["path"])
+                )
+            ):
+                record_locator = {
+                    "file_path": element["path"],
+                    "file_name": element["name"],
+                    "file_type": element["type"],
+                    "repo_path": self.connection_config.repo_path,
+                }
+                if self.connection_config.git_branch is not None:
+                    record_locator["branch"] = self.connection_config.git_branch
+
+                yield FileData(
+                    identifier=element["id"],
+                    connector_type=CONNECTOR_TYPE,
+                    source_identifiers=SourceIdentifiers(
+                        fullpath=element["path"],
+                        filename=element["path"].split("/")[-1],
+                        rel_path=rel_path,
+                    ),
+                    metadata=FileDataSourceMetadata(
+                        url=element["id"],
+                        record_locator=record_locator,
+                        permissions_data=[{"mode": element["mode"]}],
+                    ),
+                    additional_metadata={},
+                )
+
+
+class GitLabDownloaderConfig(DownloaderConfig):
+    pass
+
+
+@dataclass
+class GitLabDownloader(Downloader):
+    connection_config: GitLabConnectionConfig
+    download_config: GitLabDownloaderConfig
+
+    @SourceConnectionNetworkError.wrap
+    @requires_dependencies(["gitlab"], extras="gitlab")
+    def _fetch_content(self):
+        from gitlab.exceptions import GitlabHttpError
+
+        try:
+            project = self.connection_config.get_project()
+            content_file = project.files.get(
+                self.path,
+                ref=self.connection_config.git_branch or project.default_branch,
+            )
+        except GitlabHttpError as e:
```

We should reraise this as an error rather than returning `None` and hiding it.